backend/src/packages/chaiNNer_ncnn/ncnn/processing/upscale_image.py

```diff
@@ -138,14 +144,25 @@ def estimate_cpu():
             "Generally it's recommended to use the largest tile size possible for best performance (with the ideal scenario being no tiling at all), but depending on the model and image size, this may not be possible.",
             "If you are having issues with the automatic mode, you can manually select a tile size. On certain machines, a very small tile size such as 256 or 128 might be required for it to work at all.",
         ),
+        if_group(
+            Condition.type(1, "Image { channels: 4 } ")
+            and Condition.type(0, "NcnnNetwork { outputChannels: 1 | 3 } ")
```

Comment:
1. Welcome to python. The `and` operator will check for truthy-ness and returns the last true-y value. You meant to use bitwise AND `&`.
2. The condition is not quite correct. Since we only use this inputs if `in_nc == out_nc`, we have to use something like this:

Same for the others as well.
```suggestion
            & (Condition.type(0, "NcnnNetwork { inputChannels: 1, outputChannels: 1 }") | Condition.type(0, "NcnnNetwork { inputChannels: 3, outputChannels: 3 }"))
```